### PR TITLE
Remove `None` values from request params and data (Fixes #119)

### LINKED ISSUE #119: Implicitly remove None values from params
When calling proxmoxer API methods with `None` values, it will behave differently when using https backend vs others.

The difference comes from requests library, which will [implicitly remove `None` values from parameters](https://github.com/psf/requests/blob/1e62a3ec18e19f85ddae03b4cbbdf0b4c62834c0/requests/sessions.py#L82-L86).
On other backends the `None` values are implicitly converted to string, and providing them to the pvesh command.

A stupid, but simple example of this would be:
```python
proxmox.storage.get(type=None) # Will fail, unless https backend is used
```

Would you guys accept a PR removing `None` values just as the requests module does?

This issue surfaced from a [ansible.community issue](https://github.com/ansible-collections/community.general/issues/5449#issuecomment-1304649767) about implementing local backend.

### PR DESCRIPTION
**Pull request by kristianheljas**
In addition to cleaning up params and data, as discussed in #119 fixed data integrity checks as well.
I can separate the data integrity check fix to a separate PR if needed.

### CODE REVIEW

**Comment by jhollowe** [comment]
Yeah, separating the data integrity checks into a separate PR would be good.

**Comment by kristianheljas** [comment]
Alright, this PR is formatted now :)

I prematurely made second PR as well before noticing the pre-commit comment, gonna fix that in a jiffy.

File: proxmoxer/core.py [hunk 1]
```diff
@@ -113,6 +113,17 @@ def _request(self, method, data=None, params=None):
             logger.info("%s %s %r", method, url, data)
         else:
             logger.info("%s %s", method, url)
+
```
[the remark this answers is not in the document]
Comment (kristianheljas) [reply proxmoxer/core.py:116]: Sure, here you go :)

**Comment by jhollowe** [comment]
Thanks!

**Pull request merged**